examples/network_connectivity.py

```diff
@@ -0,0 +1,123 @@
+net = Network(params)
+
+###############################################################################
+# Instantiating the network comes with a predefined set of connections that
+# reflect the canonical neocortical microcircuit. ``net.connectivity_list``
+# is a list of dictionaries which detail every cell-cell, and drive-cell
+# connection.
+print(len(net.connectivity_list))
+print(net.connectivity_list[0])
+
+###############################################################################
+# Next we add the distal drive. Note that adding drives creates new drive-cell
+# connections that are appended to ``net.connectivity_list``
+weights_ampa_d1 = {'L2_basket': 0.006562, 'L2_pyramidal': .000007,
+                   'L5_pyramidal': 0.142300}
+weights_nmda_d1 = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
+                   'L5_pyramidal': 0.080074}
+synaptic_delays_d1 = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                      'L5_pyramidal': 0.1}
+net.add_evoked_drive(
+    'evdist1', mu=63.53, sigma=3.85, numspikes=1, weights_ampa=weights_ampa_d1,
+    weights_nmda=weights_nmda_d1, location='distal',
+    synaptic_delays=synaptic_delays_d1, seedcore=4)
+
+print(net.connectivity_list[-1])
+
+###############################################################################
+# Then we add two proximal drives and simuate the network.
+weights_ampa_p1 = {'L2_basket': 0.08831, 'L2_pyramidal': 0.01525,
+                   'L5_basket': 0.19934, 'L5_pyramidal': 0.00865}
+synaptic_delays_prox = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                        'L5_basket': 1., 'L5_pyramidal': 1.}
+# all NMDA weights are zero; pass None explicitly
+net.add_evoked_drive(
+    'evprox1', mu=26.61, sigma=2.47, numspikes=1, weights_ampa=weights_ampa_p1,
+    weights_nmda=None, location='proximal',
+    synaptic_delays=synaptic_delays_prox, seedcore=4)
+
+# Second proximal evoked drive. NB: only AMPA weights differ from first
+weights_ampa_p2 = {'L2_basket': 0.000003, 'L2_pyramidal': 1.438840,
+                   'L5_basket': 0.008958, 'L5_pyramidal': 0.684013}
+# all NMDA weights are zero; omit weights_nmda (defaults to None)
+net.add_evoked_drive(
+    'evprox2', mu=137.12, sigma=8.33, numspikes=1,
+    weights_ampa=weights_ampa_p2, location='proximal',
+    synaptic_delays=synaptic_delays_prox, seedcore=4)
+
+###############################################################################
+# Data recorded during simulations are stored under
+# :class:`~hnn_core.Cell_Response`. To test multiple network structures, we can
+# create a copy of the original network. The copied network is then simulated.
+net_ERP = net.copy()
+dpl = simulate_dipole(net_ERP, n_trials=1)
+net_ERP.cell_response.plot_spikes_raster()
+
+###############################################################################
+# We can modify the connectivity list to test the effect of different
+# connectivity patterns. For example, we can remove all layer 2 inhibitory
+# connections.
```

Comment:
almost makes me think that this PR should also come with a way to visualize the connectivity. Difficult to add? I'm thinking some kind of `matshow` plot with rows/columns grouped by cell_type

Reply:
+1!

Reply:
on second thoughts, let's bring this PR to [MRG] state first before trying to implement something new. I'm worried it will become too huge otherwise. We still have to write proper tests

Reply:
Definitely agree, that will be a good followup ENH. We'll just have to live with the agony of `plot_connectivity.py` doing no such plotting 😄